tests/testing-resources/include/aws/testing/AwsTestHelpers.h

```diff
@@ -163,14 +196,20 @@ class RetryPlanner
                     bool localResult = false;
 
                     //Try and retry a block
-                    do{}
-                    while( it->retryCount && !(localResult = it->candidate(_contextSp, it->entryId, --it->retryCount) ));
+                    do{
+                        AWS_LOGSTREAM_TRACE(ALLOCATION_TAG, "execute block="<<*it );
+                        //std::cout<<"execute block="<<*it<<std::endl;
+                    }
+                    while( it->retryCount && 
+                            !(localResult = invoker<CONTEXT>()(it->candidate, it->entryId, --it->retryCount)) &&
+                            (it->stopStrategy != StopStrategy::STOP_ON_FIRST_FAIL) 
+                         );
                     
                     //set global status from block evaluation
                     status = status && localResult;
 
                     //if stop on fail after retries, break and retry from start if available
-                    if(it->stopOnFail && !localResult)
+                    if(( it->stopStrategy == StopStrategy::STOP_ON_FIRST_FAIL || it->stopStrategy == StopStrategy::STOP_AFTER_ALL_RETRIES ) && !localResult)
```

Comment:
Added a retry plan stop strategy as needed by use case. 
One can choose to continue even on failure, stop after first failures or stop after exhausting all retries.